src/drive/web/modules/views/Upload/UploaderComponent.tsx

```diff
@@ -0,0 +1,236 @@
+
+const TypedUseI18n = useI18n as () => {
+  t: (str: string, arg?: Record<string, unknown>) => string
+}
+const TypedAlerter = Alerter as { success: (str: string) => void }
+const TypedLoadMore = LoadMore as React.FC<{
+  hasMore: boolean
+  fetchMore: () => void
+}>
+
+const DumbUpload = ({
+  onCancel,
+  uploadFilesFromNative,
+  removeFileToUploadQueue
+}: DumbUploadProps): JSX.Element | null => {
+  const [folder, setFolder] = useState<Folder>({ _id: ROOT_DIR_ID })
+  const [uploadInProgress] = useState<boolean>(false)
+  const { t } = TypedUseI18n()
+  const navigate = useNavigate()
+  const client = useClient()
+  const { items, uploadFilesFromFlagship } = useUploadFromFlagship()
+  const webviewIntent = useWebviewIntent()
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    const onFileUpload = ({
+      data: { file, isSuccess, isLast }
+    }: {
+      data: {
+        file: FileFromNative
+        isSuccess: boolean
+        isLast?: boolean
+      }
+    }): void => {
+      if (!file) return
+
+      if (!isSuccess) {
+        dispatch({ type: RECEIVE_UPLOAD_ERROR, file })
+        return
+      }
+
+      removeFileToUploadQueue({ name: file.fileName })
+        .then(() => {
+          if (isLast) {
+            return TypedAlerter.success(t('ImportToDrive.success'))
+          }
+          return
+        })
+        .catch(e => {
+          console.log('removeFileToUploadQueue error', e)
+        })
+    }
+
+    window.addEventListener('message', onFileUpload)
+  }, [dispatch, removeFileToUploadQueue, t])
+
+  const onClose = useCallback(async () => {
+    await webviewIntent?.call('resetFilesToHandle').then(() => {
+      onCancel()
+      return navigate('/')
+    })
+  }, [navigate, onCancel, webviewIntent])
+
+  const uploadFiles = useCallback(() => {
+    if (!items || items.length === 0 || !client) return
+
+    const filesForQueue = generateForQueue(items)
+    uploadFilesFromNative(
+      filesForQueue,
+      folder._id,
+      undefined,
+      { client },
+      uploadFilesFromFlagship
+    )
+      .then(() => {
+        return console.log('uploadFilesFromNative done')
+      })
+      .catch(e => {
+        console.log('uploadFilesFromNative error', e)
+      })
+
+    setTimeout(() => navigate(`/folder/${folder._id}`), 50)
+  }, [
+    items,
+    client,
+    uploadFilesFromNative,
+    folder._id,
+    uploadFilesFromFlagship,
+    navigate
+  ])
+
+  const contentQuery = buildMoveOrImportQuery(folder._id)
+  const folderQuery = buildOnlyFolderQuery(folder._id)
+
+  if (!items || items.length === 0) {
+    return null
+  }
+
+  return (
+    <FixedDialog
+      open
+      onClose={onClose}
+      size="large"
+      classes={{ paper: 'u-h-100' }}
+      title={
+        <>
+          <Header
+            entries={items}
+            title={t('ImportToDrive.title', { smart_count: items.length })}
+            subTitle={t('ImportToDrive.to')}
+          />
+          <Query
```

Comment:
Query component is deprecated. You should have refactor to useQuery(). I still thin that this kind of refactor to be up to date with the cozy latest standard should be done BEFORE converting to TS...

Reply:
refactored